### PR TITLE
[issue-90] Infoview doesn't show errors when lake fails

### PR DESCRIPTION
**Pull request by marianaalanis93**
This PR contains a fix related with #90 Infoview doesn't show errors when lake fails. In this case the message returns 'nothing' instead of undefined in some cases.

### CODE REVIEW

**Comment by gebner** [comment]
You're on the right track with the general idea.  If the `getInteractiveDiagnostics` call returns the empty array, then we should fall back to the LSP diagnostics.  That should solve the issue in #90.

However this is the wrong place to add this rule.  The `rpcInterface.ts` file contains lightweight wrappers.  These wrappers are there to 1) give the correct typescript types, and 2) register the result with the garbage collector.  They shouldn't modify the result.

The proper place to add it is in `messages.tsx`, where the `getInteractiveDiagnostics` function is called.

**Review by lovettchris — APPROVED** [review]

File: lean4-infoview/src/infoview/messages.tsx [hunk 1]
```diff
@@ -177,7 +177,8 @@ export function useMessagesForFile(uri: DocumentUri, line?: number): Interactive
             try {
                 const diags = await getInteractiveDiagnostics(rs, { uri, line: 0, character: 0 },
                     line ? { start: line, end: line + 1 } : undefined)
-                if (diags) {
+                if (diags && diags.length > 0) {
+                    // diags may be '' in some cases
```
Comment (gebner) [inline lean4-infoview/src/infoview/messages.tsx:181]:
```suggestion
                    // diags may be [] when lake fails
```

There's also another one on line 101 for the "All Messages" section.

Comment (marianaalanis93) [reply lean4-infoview/src/infoview/messages.tsx:181]: Ok checking :)

**Review by Vtec234 — APPROVED** [review]
Nice, thanks!

**Comment by Vtec234** [comment]
Thanks again! By the way, could you please follow the lean4 [commit convention](https://github.com/leanprover/lean4/blob/master/doc/dev/commit_convention.md) (which I guess we should also link somewhere in this repo)? In particular, it's better to state what is being changed in the commit title rather than what problem is that the commit solves. E.g. even something like "fix: show errors when lake fails" instead of "Infoview doesn't show errors when lake fails". You can then also link the relevant issue in the commit message like `git commit -m "fix: show errors when lake fails" -m "Fixes #90."`.

**Pull request merged**